metadata-db-app/src/cmr/metadata_db/services/concept_constraints.clj

```diff
@@ -14,15 +14,22 @@
   {:default true
    :type Boolean})
 
+(defn- find-latest-concepts
+  "Returns any concepts which match the concept-type, provider-id, and query
+  that matches the passed in concept"
+  [db provider concept query]
+  (->> query
+       (merge {:concept-type (:concept-type concept)
+               :provider-id (:provider-id concept)})
+       (c/find-latest-concepts db provider)
+       ;; Remove tombstones from the list of concepts
+       (remove :deleted)))
+
 (defn- find-latest-matching-concepts
```

Comment:
Not sure why you want to refactor this function when you could have just used it as is. Am I missing anything?